compiler/modules/front/resolveTypes.bal

```diff
@@ -28,6 +28,41 @@ function resolveTypes(ModuleSymbols mod) returns ResolveTypeError? {
     foreach var defn in mod.defns {
         check resolveDefn(mod, defn);
     }
+    check finishDeferredEmptinessChecks(mod);
+}
+
+function finishDeferredEmptinessChecks(ModuleSymbols mod) returns ResolveTypeError? {
+    if !mod.tc.env.isReady() {
+        panic err:impossible("type environment is not ready");
+    }
+    foreach var { semType, modDefn, td } in mod.deferredEmptinessChecks {
+        // XXX When we can give multiple errors we should check all the deferred emptiness checks
+        check finishEmptinessCheck(mod, semType, modDefn, td);
+    }
+    foreach var [checked, loc] in mod.recursiveEmptinessChecked {
+        if !checked {
```

Comment:
Isn't the first half of the tuple saying whether it's recursive? Everything in the map is checked.

Reply:
Yes. Changed to `recursive` in the `AtomicEmptyType`